packages/plugins/plugin-mixpanel/src/MixpanelPlugin.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  DestinationPlugin,
+  PluginType,
+  TrackEventType,
+  ScreenEventType,
+  SegmentAPISettings,
+  UpdateType,
+  SegmentMixpanelSettings,
+  IdentifyEventType,
+  GroupEventType,
+  JsonMap,
+  AliasEventType,
+} from '@segment/analytics-react-native';
+import { Mixpanel } from 'mixpanel-react-native';
+import identify from './methods/identify';
+import screen from './methods/screen';
+import group from './methods/group';
+import alias from './methods/alias';
+import mixpanelTrack from './methods/mixpanelTrack';
+
+export class MixpanelPlugin extends DestinationPlugin {
+  type = PluginType.destination;
+  key = 'Mixpanel';
+  trackScreens = false;
+  private mixpanel: Mixpanel | undefined;
+  private settings: SegmentMixpanelSettings | null = null;
+
+  update(settings: SegmentAPISettings, _: UpdateType) {
+    const mixpanelSettings = settings.integrations[
+      this.key
+    ] as SegmentMixpanelSettings;
+
+    if (!mixpanelSettings || this.mixpanel !== undefined) {
+      return;
+    }
+    if (mixpanelSettings.token.length) {
+      this.mixpanel = new Mixpanel(mixpanelSettings.token);
+      this.mixpanel.init();
+      this.settings = mixpanelSettings;
+    } else {
+      return;
+    }
+
+    if (mixpanelSettings.enableEuropeanEndpoint) {
+      this.mixpanel?.setServerURL('api.eu.mixpanel.com');
+    }
+  }
+
+  identify(event: IdentifyEventType) {
+    identify(event, this.mixpanel!, this.settings!);
```

Comment:
I don't think it is safe to force unwrap `this.mixpanel` and `this.settings` since they get assigned by the update only when the settings were received.

I think it should be handled here or in each method explicitely:
```
if (this.mixpanel !== undefined && this.settings !== undefined) {
  identify(event, this.mixpanel, this.settings)
}
return event
```

Reply:
(Same for all the other methods)